sql/catalyst/src/main/scala/org/apache/spark/sql/sources/filters.scala

```diff
@@ -32,6 +32,7 @@ import org.apache.spark.annotation.{Evolving, Stable}
 sealed abstract class Filter {
   /**
    * List of columns that are referenced by this filter.
+   * Note that, if a column contains `dots` in name, it will be quoted to avoid confusion.
```

Comment:
Do we have an assumption that these column names match with the original source names already? 
Suddenly, I'm wondering if this is safe for all data sources like JDBC? Some DBMS like PostgreSQL is case-sensitive.

Reply:
Parquet has some logics to handle what you described. I was wondering if it's safe for ORC as well.

Reply:
Okay, I was confused about the reviewing order. This will be a breaking change - I noted in https://github.com/apache/spark/pull/27780#discussion_r389331398 too. There are non-DBMS sources too.

Reply:
Is this a behavior change or we just clarify it in the comment?

Reply:
When there is a `dot` in the column name,  the existing `Filter` will not quote it. As a result, to extend it to support nested column, we need to quote the `dot` in name to distinguish from the `dot` as a separator of nested column.

The behavior change is only for the column name that contains `dot`.